packet_link_qualification/packet_link_qualification.proto

```diff
@@ -70,98 +78,92 @@ enum QualificationState {
-  enum QualificationEnd {
-    END_UNSPECIFIED = 0;
-    // Responsible for generating loopback packets, etc.
-    NEAR_END = 1;
-    // Responsible for looping back packets only.
-    FAR_END = 2;
-  }
-
-  message QualificationConfiguration {
-    // The ID assigned to the given test. The ID value must be unique among the
-    // tests running on this agent.
-    string id = 1;
-
-    // Path to the interface corresponding to the port to test.
-    .gnoi.types.Path interface = 2;
-
-    // The minimum amount of wait time before sending packets on the link. It
-    // gives peer additional time to prepare. The default value for this field
-    // is 70 seconds.
-    int32 minimum_wait_before_preparation_seconds = 3;
-
-    // The total number of packets to send on the link in the preparation phase.
-    int64 num_preparation_packets = 4;
-
-    // The amount of time in which the preparation must complete.
-    int32 preparation_timeout_seconds = 5;
-    
-    // The amount of time the qualification will run before we recapture the
-    // packets and check if there was any drop.
-    int32 qualification_duration_seconds = 6;
-    
-    // Qualification end.
-    QualificationEnd qualification_end = 7;
-    
-    // The default value for this field is 10 seconds. This is the 
-    // time period that the link qualification service should wait 
-    // between the two samples of port counters that it reads from the 
-    // device - both prior to starting qualification, and right after 
-    // qualification. The qualification itself runs for 
-    // Qaulification_duration_seconds.
-    int32 test_wait_between_sample_seconds = 8;
-    
-    // The default value of this field is 10 seconds. During this period, the 
-    // link qualification state machine must undo things it setup during start 
-    // of the test and ensure all packets circulating as part of the test have
-    // been dropped.
-    int32 minimum_wait_post_qualification_seconds = 9;
-    
-    // default packet_size is 1500 bytes.
-    uint16 packet_size = 10;
-    
-  }
+enum QualificationEnd {
+  END_UNSPECIFIED = 0;
+  // Responsible for generating loopback packets, etc.
+  END_NEAR = 1;
+  // Responsible for looping back packets only.
+  END_FAR = 2;
+}
 
+message StartPacketQualificationRequest {
   // Per interface start configuration request.
-  repeated QualificationConfiguration configs = 1;
+  repeated QualificationConfiguration config = 1;
+}
+
+message QualificationConfiguration {
+  // The ID assigned to the given test. The ID value must be unique among the
+  // tests running on this agent.
+  string id = 1;
+
+  // Path to the interface corresponding to the port to test.
+  types.Path interface = 2;
+
+  // The minimum amount of wait time before sending packets on the link. It
+  // gives peer additional time to prepare. The default value for this field
+  // is 70 seconds.
+  int32 minimum_wait_before_preparation_seconds = 3;
```

Comment:
after thinking on this more if our expectation is that the device after start may become unreachable should we break this into a setup step then a start step this would let you setup both sides before starting both sides
where the setup could validate that device capable of doing the work vs. the start setting up and "waiting to start" then the other side try to setup and fails then you have "quickly" jump back to the peer. device and stop